### ISSUE TITLE
Conflicting groovy dependencies

### ISSUE THREAD

**Issue by centosGit** [JanusGraph/janusgraph#407]
Hi

I am following the tutorials to setup a standalone gremlin server and then install janus graph dependencies as shown in the documentation. Installation goes fine but when i start the gremlin server i get exceptions on the groovy libraries pre installed with gremlin.

`Caused by: groovy.lang.GroovyRuntimeException: Conflicting module versions. Module [groovy-jsr223 is loaded in version 2.4.11 and you are trying to load versio│
n 2.4.7`

Basically the easiest solution is to 
`rm lib/groovy-*`

Is this to be expected? 

Here is my grape config
`
<ivysettings>
  <settings defaultResolver="downloadGrapes"/>           
  <resolvers>
    <chain name="downloadGrapes">
      <filesystem name="cachedGrapes">
        <ivy pattern="${user.home}/.groovy/grapes/[organisation]/[module]/ivy-[revision].xml"/> 
        <artifact pattern="${user.home}/.groovy/grapes/[organisation]/[module]/[type]s/[artifact]-[revision].[ext]"/>
      </filesystem>                    
      <ibiblio name="codehaus" root="http://repository.codehaus.org/" m2compatible="true"/>                                                                    
      <ibiblio name="central" root="http://central.maven.org/maven2/" m2compatible="true"/>                                                                    
      <ibiblio name="jitpack" root="https://jitpack.io" m2compatible="true"/>  
      <ibiblio name="java.net2" root="http://download.java.net/maven/2/" m2compatible="true"/>                                                                 
          <ibiblio name="oracle" root="http://download.oracle.com/maven/" m2compatible="true"/>                                                                
    </chain>                           
  </resolvers>                         
</ivysettings>                         
`

Besides the grape config none of the configuration is modified at this point. If i remove the groovy libs then it uses the other version and works fine.

**Comment by pluradj** [comment]
You should make sure you match the Apache TinkerPop version with the version JanusGraph uses. JanusGraph 0.1.1 uses TinkerPop 3.2.3.

I'm curious to hear why you'd want to install JanusGraph into a standalone Gremlin Server. Unzipping the JanusGraph distributable gives you a working Gremlin Server out of the box with little fuss. It actually might make more sense to remove that section of the documentation.

**Comment by robertdale** [comment]
And JG 0.2-SNAPSHOT uses TinkerPop 3.2.5 (which is where groovy 2.4.11 was introduced)

**Comment by robertdale** [comment]
@pluradj I'm not sure what the tutorial is. Is it different than the section under [Advanced JanusGraph Server Configurations](http://docs.janusgraph.org/latest/server.html#_advanced_janusgraph_server_configurations) ?
  In any case, it is applicable to a deployment model of  [Remote Server Mode with Gremlin Server](http://docs.janusgraph.org/latest/cassandra.html#_remote_server_mode_with_gremlin_server) so there should remain some directions on how to do it.

**Comment by pluradj** [comment]
Yeah, I believe @centosGit is referring to the docs under 7.4.2 ["Using TinkerPop Gremlin Server with JanusGraph"](http://docs.janusgraph.org/latest/server.html#gremlin-server-with-janusgraph) where it tells you to install `janusgraph-all` as a Gremlin Server plugin.

By unzipping the `janusgraph-0.1.1-hadoop2.zip`, you already have an environment ready for remote server mode with Gremlin Server. The required libraries that you would be installing with `janusgraph-all` are already under `$JANUSGRAPH_HOME/lib`.

The docs under 7.2 ["JanusGraph Server as a WebSocket Endpoint"](http://docs.janusgraph.org/latest/server.html#_janusgraph_server_as_a_websocket_endpoint) talk a little bit about this. So I'm thinking perhaps we need to clean up these docs to make it more clear.

**Comment by centosGit** [comment]
@pluradj  Yes that is exactly the section in the documentation that i was following. I got the impression that a production setup should follow that approach( setting up a stand alone gremlin server and pull janus graph dependencies ). 

My bad. I was using tinkerpop 3.2.5 and not 3.2.3. I will close the issue.

Just a clarification. Are you saying that i should not bother with the approach i followed? Should i just use the janus graph and even deploy like that? I only thought that the way janus is distributed and bundled in that way was for convenience.

**Comment by pluradj** [comment]
We should leave this open for updating the docs. I'm not clear what the use case would be to pull Gremlin Server from the Apache downloads site and then install JanusGraph into it. I guess one use case would be to configure a server with only the jars required for a specific backends. If you look in the zip file distribution's `lib` directory, you'll see dependencies for all the supported backends in there. Of course, installing `janusgraph-all` would end up doing the same. It would make more sense just to install the specific components you need.

`bin/janusgraph.sh` is a convenience script that starts a single Cassandra node, a single Elasticsearch node, and the Gremlin Server. As described in 7.1.1 ["Using the Pre-Packaged Distribution"](http://docs.janusgraph.org/latest/server.html#_using_the_pre_packaged_distribution):

> The quick-start bundle is not intended to be representative of a production installation

https://github.com/JanusGraph/janusgraph/issues/371 is a related ticket.

@twilmes any thoughts to add here on Gremlin Server deployments?

**Comment by centosGit** [comment]
@pluradj Thanks for the answer. I will change my approach and start using the janus graph distribution as is.
